single_kernel_mongo/managers/config.py

```diff
@@ -332,16 +332,20 @@ def audit_options(self) -> dict[str, Any]:
     def auth_parameter(self) -> dict[str, Any]:
         """The auth mode."""
         cmd = {"security": {"authorization": "enabled"}} if self.auth else {}
-        if self.state.tls.internal_enabled and self.state.tls.external_enabled:
+        if self.state.tls.peer_enabled:
             return always_merger.merge(
                 cmd,
                 {
                     "security": {"clusterAuthMode": "x509"},
                     "net": {
                         "tls": {
+                            "mode": "preferTLS",
```

Comment:
Didn't we go with `requireTLS` in a previous PR?

Reply:
We talked about it: if you have only internal, you must use `preferTLS` otherwise all clients will be mandated to use TLS relations (which is not what we want).
MongoDB has only one setting for TLS mode that works for both internal and external, with the following values and semantics:
```
disabled: The server does not use TLS.
allowTLS: Connections between servers do not use TLS. For incoming connections, the server accepts both TLS and non-TLS.
preferTLS: Connections between servers use TLS. For incoming connections, the server accepts both TLS and non-TLS.
requireTLS: The server uses and accepts only TLS encrypted connections.
```